Sources/XiEditor/AppDelegate.swift

```diff
@@ -285,6 +264,38 @@ class AppDelegate: NSObject, NSApplicationDelegate {
     }
 
     //- MARK: - helpers
+    func setupConfigDirectory() {
+        let applicationDirectory = FileManager.default.urls(
+            for: .applicationSupportDirectory,
+            in: .userDomainMask)
+            .first!
+            .appendingPathComponent("XiEditor")
+        let preferencesDirectory = applicationDirectory.appendingPathComponent(PREFERENCES_FILE_NAME)
+
+        // Create application support directory and copy preferences file first, then
+        // copies the example config to `preferences.xiconfig` if it isn't found.
+        if !FileManager.default.fileExists(atPath: applicationDirectory.path) {
+            do {
+                try FileManager.default.createDirectory(at: applicationDirectory,
+                                                        withIntermediateDirectories: true,
+                                                        attributes: nil)
+                let preferencesPath = applicationDirectory.appendingPathComponent(PREFERENCES_FILE_NAME)
+                let defaultConfigPath = Bundle.main.url(forResource: "client_example", withExtension: "toml")
+                try FileManager.default.copyItem(at: defaultConfigPath!, to: preferencesPath)
+            } catch let err  {
+                fatalError("Failed to create application support directory \(applicationDirectory.path). \(err)")
+            }
+        } else if !FileManager.default.fileExists(atPath: preferencesDirectory.path) {
+            do {
+                let defaultConfigPath = Bundle.main.url(forResource: "client_example", withExtension: "toml")
```

Comment:
This block duplicates code above. Instead of having this be two separate cases, can't we just have to separate `if` statements, first checking for the directory (creating if needed) and then checking for the file, also creating if needed?